### ISSUE TITLE
ListTile.divideTiles throws an error if the tiles list is empty

### ISSUE THREAD

**Issue by miquelbeltran** [flutter/flutter#75518]
Calling to `ListTile.divideTiles` with an empty list of tiles will cause an error when null safety is enabled.

## Steps to Reproduce

1. Create a project with null safety enabled.
2. Call to `final output = ListTile.divideTiles(context: context, tiles: [] );` (notice the empty list)
3. Call to `print(output);`
4. An exception is thrown: `type 'Null' is not a subtype of type 'Widget' in type cast`

Here's a minimal example that reproduces the issue:

```
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final output = ListTile.divideTiles(context: context, tiles: []);
    print(output);
    return Container();
  }
}
```

Important, in the pubspec.yaml to enable null safety:

```
environment:
  sdk: ">=2.12.0-0 <3.0.0"
```

As well, this issue is reproducible in the codelab Write your first Flutter app part 2 **when null safety is enabled**: https://codelabs.developers.google.com/codelabs/first-flutter-app-pt2#0

**Expected results:** 

The output should be an empty list (of widgets).

**Actual results:**

The call throws an exception.

<details>
  <summary>Logs</summary>

```
======== Exception caught by widgets library =======================================================
type 'Null' is not a subtype of type 'Widget' in type cast
The relevant error-causing widget was: 
  MyApp file:///home/miquel/tmp/listtile_error/lib/main.dart:4:10
====================================================================================================

======== Exception caught by widgets library =======================================================
The following _CastError was thrown building MyApp(dirty):
type 'Null' is not a subtype of type 'Widget' in type cast

The relevant error-causing widget was: 
  MyApp file:///home/miquel/tmp/listtile_error/lib/main.dart:4:10
When the exception was thrown, this was the stack: 
#0      ListIterator.current (dart:_internal/iterable.dart:328:29)
#1      ListTile.divideTiles.sync_op (package:flutter/src/material/list_tile.dart:995:28)
#2      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:181:26)
#3      _iterablePartsToStrings (dart:collection/iterable.dart:318:13)
#4      IterableBase.iterableToShortString (dart:collection/iterable.dart:238:7)
...
====================================================================================================

```

```
Analyzing listtile_error...                                             
No issues found! (ran in 0.8s)

```

```
flutter doctor -v
[✓] Flutter (Channel beta, 1.26.0-17.3.pre, on Linux, locale en_US.UTF-8)
    • Flutter version 1.26.0-17.3.pre at /home/miquel/dev/tools/flutter
    • Framework revision 4b50ca7f7f (14 hours ago), 2021-02-04 19:44:27 -0800
    • Engine revision 2c527d6c7e
    • Dart version 2.12.0 (build 2.12.0-259.8.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/miquel/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /usr/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_275-8u275-b01-0ubuntu1~20.04-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[!] Android Studio (not installed)
    • Android Studio not found; download from https://developer.android.com/studio/index.html
      (or visit https://flutter.dev/docs/get-started/install/linux#android-setup for detailed instructions).

[✓] IntelliJ IDEA Ultimate Edition (version 2020.3)
    • IntelliJ at /home/miquel/.local/share/JetBrains/Toolbox/apps/IDEA-U/ch-0/203.6682.168
    • Flutter plugin version 53.0.5
    • Dart plugin version 203.6912

[✓] Connected device (2 available)
    • sdk gphone x86 (mobile) • emulator-5554 • android-x86    • Android 11 (API 30) (emulator)
    • Chrome (web)            • chrome        • web-javascript • Google Chrome 88.0.4324.96

! Doctor found issues in 1 category.

```

</details>

**Comment by miquelbeltran** [comment]
I will provide a PR with a fix for this shortly.

**Comment by pedromassangocode** [comment]
```
======== Exception caught by widgets library =======================================================
The following _CastError was thrown building MyApp(dirty):
type 'Null' is not a subtype of type 'Widget' in type cast

The relevant error-causing widget was: 
  MyApp file:///Users/pedromassango/dev/Projects/flutter_example/lib/main.dart:4:10
When the exception was thrown, this was the stack: 
#0      ListIterator.current (dart:_internal/iterable.dart:328:29)
#1      ListTile.divideTiles.sync_op (package:flutter/src/material/list_tile.dart:995:28)
#2      _SyncIterator.moveNext (dart:core-patch/core_patch.dart:181:26)
#3      _iterablePartsToStrings (dart:collection/iterable.dart:318:13)
#4      IterableBase.iterableToShortString (dart:collection/iterable.dart:238:7)
...
```

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel beta, 1.26.0-17.3.pre, on macOS 11.2 20D64 darwin-x64, locale en-AO)
    • Flutter version 1.26.0-17.3.pre at /Users/pedromassango/Code/flutter_beta
    • Framework revision 4b50ca7f7f (4 days ago), 2021-02-04 19:44:27 -0800
    • Engine revision 2c527d6c7e
    • Dart version 2.12.0 (build 2.12.0-259.8.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.2.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 88.0.4324.150

! Doctor found issues in 1 category.
```

</details>

**Comment by miquelbeltran** [comment]
Thanks for confirming @pedromassango !

The fix is ready in the other PR and waiting to be merged then the tree is green.

**Comment by miquelbeltran** [comment]
Fix is now merged to master

**Comment by RobertHeim** [comment]
Isn't this released? At least I still experience this in 2.0.6 ?
